contracts/VaultFactory.sol

```diff
@@ -2,11 +2,14 @@
 
 pragma solidity ^0.6.12;
 
+import "@openzeppelin/contracts/access/AccessControl.sol";
 import "./lib/ProxyFactory.sol";
 import "./interface/IBazrToken.sol";
 
-contract VaultFactory is ProxyFactory {
+contract VaultFactory is ProxyFactory, AccessControl {
 
+  bytes32 public constant ADMIN = keccak256("ADMIN_ROLE");
```

Comment:
you're right, not necessary since we have only one role right now. In the future we can be explicit with the role names with this since the constant is the encoded string of the role name. `keccak256("MINTER_ROLE")`, `keccak256("ADMIN_ROLE")` etc. But for MVP the provided DEFAULT_ADMIN_ROLE should be sufficient, we just have to record or remember that this role is encoded as `0x00`. We need to know the encoding to pass onto functions such as `grantRole` and `hasRole`